### PR TITLE
Add react-dom-unstable-native-dependencies

### PR DESCRIPTION
**Pull request by fmoo**
react-native-web and react-primitives currently access a few internals
for shimming DOM events into native ones.  Changes in react@16 packaging
hide these internals, which has made it difficult for these projects to upgrade.
change adds a submodule to react-dom, unstable-native-dependencies that
includes the necessary modules to continue enabling that method of
dom-native event injection.

The bundle config is currently a bit of dogscience.  The one thing I really changes were:

- basically copied it from react-dom
- added an externals / globals alias for react-dom in order to import it from inside the module
which was required to inject the shared ComponentTree into this module.
- removed the fb entry point since I wasn't sure what it did or if it would be necessary

### CODE REVIEW

**Comment by gaearon** [comment]
>removed the fb entry point since I wasn't sure what it did or if it would be necessary

It would be useful if we wanted to use these modules in www too.

File: packages/react-dom/unstable-native-dependencies.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+'use strict';
```
Comment (gaearon) [inline packages/react-dom/unstable-native-dependencies.js]: Since there's a new file we need to add it to whitelist in package.json, or it would get ignored on publish.

File: scripts/rollup/bundles.js [hunk 2]
```diff
@@ -125,6 +125,34 @@ const bundles = [
       'src/shared/**/*.js',
     ],
   },
+  /* React DOM internals required for shimming react-native apps into the react-dom renderer */
```
Comment (gaearon) [inline scripts/rollup/bundles.js]: Maybe let's say "required for react-native-web" for explicitness.

**Comment by gaearon** [comment]
Is there a smoke test we can add? Otherwise I'm worried we'll break it.

**Comment by fmoo** [comment]
Still trying to figure out the fbEntry business.  When I add it and try to build, I get the following unhandled error when trying to build:

```
'ReactDOM' is imported by src/renderers/dom/shared/ReactDOMUnstableNativeDependenciesEntry.js,
but could not be resolved – treating it as an external dependency
```
Any ideas?

**Comment by gaearon** [comment]
Please `yarn prettier`, thanks!

**Comment by gaearon** [comment]
LGTM, thank you for driving this!

**Comment by necolas** [comment]
I really appreciate this. Thank you!
